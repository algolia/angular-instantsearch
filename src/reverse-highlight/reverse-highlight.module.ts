import { NgModule } from '@angular/core';

import { NgAisReverseHighlight } from './reverse-highlight';

@NgModule({
  declarations: [NgAisReverseHighlight],
  exports: [NgAisReverseHighlight],
})
export class NgAisReverseHighlightModule {}
