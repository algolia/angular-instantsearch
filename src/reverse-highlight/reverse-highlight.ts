import { ChangeDetectionStrategy, Component, Input } from '@angular/core';

import { Hit } from 'instantsearch.js';
import { reverseHighlight } from 'instantsearch.js/es/helpers';

@Component({
  selector: 'ais-reverse-highlight',
  changeDetection: ChangeDetectionStrategy.OnPush,
  template: `<span class="ais-ReverseHighlight" [innerHtml]="content"></span>`,
})
export class NgAisReverseHighlight {
  @Input() private readonly attribute: string;
  @Input() private readonly hit: Partial<Hit>;
  @Input() private readonly highlightedTagName: string = 'mark';

  get content() {
    return reverseHighlight({
      attribute: this.attribute,
      hit: this.hit,
      highlightedTagName: this.highlightedTagName,
    });
  }
}
