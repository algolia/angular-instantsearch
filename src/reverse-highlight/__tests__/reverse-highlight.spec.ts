import { Component } from '@angular/core';
import { TestBed } from '@angular/core/testing';

import { NgAisReverseHighlightModule } from '../reverse-highlight.module';

const render = ({ hit, attribute, highlightedTagName = 'mark' }) => {
  @Component({
    selector: 'test-component',
    template: `
      <ais-reverse-highlight [attribute]="attribute" [hit]="hit" [highlightedTagName]="highlightedTagName">
      </ais-reverse-highlight>
    `,
  })
  class TestComponent {
    hit = hit;
    attribute: string = attribute;
    highlightedTagName = highlightedTagName;
  }

  TestBed.configureTestingModule({
    declarations: [TestComponent],
    imports: [NgAisReverseHighlightModule],
  });

  const fixture = TestBed.createComponent(TestComponent);
  fixture.detectChanges();

  return fixture;
};

describe('reverse highlight', () => {
  it('should highlight strings', () => {
    const fixture = render({
      attribute: 'name',
      hit: {
        _highlightResult: {
          name: { value: '<mark>foo</mark> bar' },
        },
      },
    });
    expect(fixture).toMatchSnapshot();
  });

  it('should highlight with a custom highlightedTagName', () => {
    const fixture = render({
      attribute: 'name',
      hit: {
        _highlightResult: {
          name: { value: '<mark>foo</mark> bar' },
        },
      },
      highlightedTagName: 'em',
    });
    expect(fixture).toMatchSnapshot();
  });
});
