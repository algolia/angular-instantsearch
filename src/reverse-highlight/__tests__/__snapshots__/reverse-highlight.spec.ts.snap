// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reverse highlight should highlight strings 1`] = `
<test-component
  attribute={[Function String]}
  highlightedTagName={[Function String]}
  hit={[Function Object]}
>
  <ais-reverse-highlight>
    <span
      class="ais-ReverseHighlight"
    >
      foo
      <mark
        class="ais-ReverseHighlight-highlighted"
      >
         bar
      </mark>
    </span>
  </ais-reverse-highlight>
</test-component>
`;

exports[`reverse highlight should highlight with a custom highlightedTagName 1`] = `
<test-component
  attribute={[Function String]}
  highlightedTagName={[Function String]}
  hit={[Function Object]}
>
  <ais-reverse-highlight>
    <span
      class="ais-ReverseHighlight"
    >
      foo
      <em
        class="ais-ReverseHighlight-highlighted"
      >
         bar
      </em>
    </span>
  </ais-reverse-highlight>
</test-component>
`;
